Lib/fractions.py

```diff
@@ -156,10 +141,13 @@ def __new__(cls, numerator=0, denominator=None, *, _normalize=True):
                             denominator *= 10**-exp
                 if m.group('sign') == '-':
                     numerator = -numerator
-
             else:
-                raise TypeError("argument should be a string "
-                                "or a Rational instance")
+                try:
+                    self._numerator, self._denominator = math._as_integer_ratio(numerator)
+                    return self
+                except TypeError:
+                    raise TypeError("argument should be a string or a number, "
```

Comment:
Since the name of the encompassing type is "numeric" rather than "number", can we adjust the error message?
```suggestion
                    raise TypeError("argument type should be string or numeric, "
```
    raise TypeError("Argument type should be string or numeric, "


Source: https://docs.python.org/3.9/library/stdtypes.html#numeric-types-int-float-complex

Reply:
But an instance of a numeric type is a number, is not?

And if use "argument type", it should be "str", not "string".

Reply:
> But an instance of a numeric type is a number, is not?

It is, but I think that it's a bit more useful to users to specify the actual types in this case since it's a ``TypeError``. Also, when searching the docs for "number", the relevant documentation page ("Built-in Types") does not come up as a suggestion, instead they'll likely encounter the page for the "numbers" module.  When searching for "numeric", more relevant results are found, including "Built-in Types". 

> And if use "argument type", it should be "str", not "string".

I'll update the suggestion accordingly.

Reply:
This is a standard message used in many sites. For example:
```
>>> float([])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: float() argument must be a string or a number, not 'list'
```
If you want to change it, please open a separate issue.

Reply:
> If you want to change it, please open a separate issue.

Alright, I'll take a look at what other areas use this error message and consider whether or not it should be addressed. That would definitely go outside of the scope of this PR.